vector/src/main/scala/geotrellis/vector/affine/AffineTransformation.scala

```diff
@@ -22,7 +22,7 @@ trait AffineTransformation {
   def transform(geom: MultiLine): MultiLine = MultiLine(transform(geom.jtsGeom))
   def transform(geom: MultiPolygon): MultiPolygon = MultiPolygon(transform(geom.jtsGeom))
   def transform(geom: GeometryCollection): GeometryCollection = GeometryCollection(transform(geom.jtsGeom))
-  def transform(geom: Geometry): Geometry = Geometry(transform(geom.jtsGeom))
+  def transform(geom: Geometry): Geometry = Geometry[Geometry](transform(geom.jtsGeom))
```

Comment:
This call looks a bit awkward...maybe a reason to not use apply and have an explicit method call?

```scala
def transform(geom: Geometry): Geometry =
  Geometry.wrap[Geometry](transform(geom.jtsGeom))
```

also we have implicit conversions for JTS geometries to wrapped types, except for the base class for some reason. e.g. https://github.com/dwins/geotrellis/blob/fix-geometry-apply/vector/src/main/scala/geotrellis/vector/Point.scala#L29

If we did have this conversion for Geometry, this could be written as 

```scala
def transform(geom: Geometry): Geometry =
  transform(geom.jtsGeom)
```

although, I think being explicit is better in this case. Even if it's not used here it would be good to add that implicit conversion to match the rest of the geometries.

I'm not really in love with any of these, but leaning towards `Geometry.wrap`. Thoughts?